src/app/components/LiveLabelPromo/index.tsx

```diff
@@ -0,0 +1,32 @@
+/** @jsx jsx */
+/** @jsxRuntime classic */
+/* @jsxFrag React.Fragment */
+import React, { PropsWithChildren } from 'react';
+import { jsx } from '@emotion/react';
+import LiveLabel from '../LiveLabel';
+import { LiveLabelProps } from '../LiveLabel/types';
+import styles from './index.styles';
+
+const LiveLabelPromo = ({
+  lang = 'en-GB',
+  id,
+  children,
+  offScreenText,
+  className,
+}: PropsWithChildren<LiveLabelProps>) => {
+  const pulse = [
+    styles.liveLabelPulse,
+    className === 'first-promo' && styles.firstPromo,
+  ];
+
+  return (
+    <>
+      <LiveLabel.Pulse className={className} css={pulse} />
+      <LiveLabel.Text lang={lang} id={id} offScreenText={offScreenText}>
+        {children}
+      </LiveLabel.Text>
+    </>
+  );
+};
+
+export default LiveLabelPromo;
```

Comment:
I am wondering if we could agree on a better name for this - it's not a Promo, it's just the Pulse & Label together... (I can't think of a good name right now, but I know that LiveLabel kinda implies it's a label only, and not a pulse 🤦‍♀️ ). `LiveComponent`? Nope! `LiveIndicator`?

Reply:
Yeah just came up with a quick name. I think the parent compound component should be called `LiveIndicator`, but the applied one is used on promos, hence why I named itthat way, `LiveIndicatorForPromos` maybe? but that looks weird as well

Reply:
Or perhaps `LivePromoIndicator`?

Reply:
Actually, thinking about this a bit more - let's not even add the concept of Promo at all. I think the parent compound component should be called `LiveIndicator`, and the other components called `LivePulse` and `LiveText`. Our Homepages will use the `LiveIndicator`, while the LivePage will use the `LivePulse` + `LiveText` components with additional styling applied.

Reply:
So the [parent compound component](https://github.com/bbc/simorgh/blob/5e7e2d95ec5bc809be12bfe8b4221c9e657ecd90/src/app/components/LiveLabel/index.tsx#L7-L22), acts as an entry point for importing `Pulse` and `Text`, it does nothing with the layout of those components, and handles any state needed for them. Could it get confusing if we turn them into separate components?

Are you suggesting something like this?:
```
- components
-- LiveIndicator (Used throughout)
-- LivePulse
-- LiveText
```
Instead of:
```
- components
-- LiveIndicator
--- Pulse
--- Text
-- LiveLabelPromo (Used throughout)
```

Reply:
If the first option works, I would prefer that one (although would welcome a second opinion if anyone had one? And I agree about the potential complications of having lots of components) I am not sure what this does for your implementation - e.g. do you still need the LiveLabelContext?

Reply:
I think there's a few gaps in our component standards around how these compound components are organised. 

For example in the OptimoPromos, we add the [service to the compound components context](https://github.com/bbc/simorgh/blob/6fe88689a7cb76f8afe67bed7338ee9da1c16297/src/app/legacy/components/OptimoPromos/index.jsx#L24). Should we do that in this instance? Confusingly the OptimoPromos also imports the [service context](https://github.com/bbc/simorgh/blob/6fe88689a7cb76f8afe67bed7338ee9da1c16297/src/app/legacy/components/OptimoPromos/Content/index.jsx#L10-L12) in child components as well. It's also confusing as it houses components which [live somewhere separately or are used outside of the OptimoPromos](https://github.com/bbc/simorgh/blob/6fe88689a7cb76f8afe67bed7338ee9da1c16297/src/app/legacy/components/OptimoPromos/index.jsx#L43-L46).

Then on the need for `LiveLabelContext`, all compound patterns we've used so far have context involved, [OptimoPromos](https://github.com/bbc/simorgh/blob/6fe88689a7cb76f8afe67bed7338ee9da1c16297/src/app/legacy/components/OptimoPromos/index.jsx#L28-L30) and [Promo](https://github.com/bbc/simorgh/blob/6fe88689a7cb76f8afe67bed7338ee9da1c16297/src/app/legacy/components/Promo/index.jsx#L38), does that mean we ditch the pattern when context isn't needed? Or do we need context because that's where we pass down other contexts like request and service for example? Also does it get confusing for devs if they a see a compound component with no context boilerplate? The [article](https://kentcdodds.com/blog/compound-components-with-react-hooks) we link to for this pattern always has context in it's examples.

Reply:
Yea I don't think we should be passing `service` from ServiceContext down through _another_ context. ServiceContext is available throughout the entire application in any renderable component, so I don't see the need in passing it as a prop.

Similarly, as you say we shouldn't reference external components as part of a compound component. Fairly dangerous as removal of those external components would then break the compound component that references them.

Reply:
I'd say for the `LiveLabelContext`, or any context, to only use it if its necessary and helps reduce the amount of state you're passing around.

If `LiveLabelContext` isn't used then I'd remove it and add it only when we need to.

Reply:
Just a quick question/clarification on 
> we shouldn't reference external components as part of a compound component. Fairly dangerous as removal of those external components would then break the compound component that references them.

Do you mean that the compound - in this case - LiveIndicator - should not reference external / sibling components? We would end up with broken compound components if someone were to remove primitive components such as `Image`, `Timestamp` etc, which live in the src/app/components directory. So I am not sure if we need to be concerned about that in this case? I think any of us would check references to components before making changes. We also have visual snapshot tests (Chromatic) to make sure that our components are still rendering as expected.

Or do you think that the second option would avoid this situation, even if we did end up with some duplication?

I think we will need to be really clear in the usage docs for all of these components, so that people know which component to use, when, and in which context (not the `context` we're speaking about ☝️ 😅 )

Reply:
I mean in Harveys example of the OptimoPromo https://github.com/bbc/simorgh/blob/6fe88689a7cb76f8afe67bed7338ee9da1c16297/src/app/legacy/components/OptimoPromos/index.jsx#L43-L46, I feel we shouldn't be exposing `Promo.Timestamp = Timestamp;` as anyone consuming the `Promo.Timestamp` would rightfully assume that it comes from `OptimoPromo` since its exported from there, but in reality it isn't.

Using external primitive components within compound components and its siblings is fine, but exposing an external component as a compound component sibling isn't imo.

Reply:
In that example, we could have created a local `Timestamp` component and imported the Psammead Timestamp into that and then exported it as `Promo.Timestamp = Timestamp`. 

Its a very light/thin wrapper around a component, but its clearer what its doing in that case. Otherwise we have a very tight dependency lock situation like we have there.

Reply:
> In that example, we could have created a local `Timestamp` component and imported the Psammead Timestamp into that and then exported it as `Promo.Timestamp = Timestamp`.

Sounds like this could as part of the impactful promo work if OptimoPromos is used

Reply:
Anyway @karinathomasbbc, I've moved the layout logic into the parent compound component. I've decided not to rename it in this PR, as I didn't want to make it too bloated to review as there are a lot of references to live label in documentation and the code, and it allows us to move this along a bit faster. We can open up a new ticket if we've got a strong opinion on changing it. 

I've kept the pulse and text within the parent compound component, as I think co-location is better, but I've kept the LiveLabel.Pulse and LiveLabel.Text exports, meaning we can import LiveLabel directly and reference those child components on the live pages. 

I've kept a note of some of the things we've discussed so if we do another component standards session to clarify some thing around components, we can bring them up in the meeting and update our docs

Reply:
Sounds good, thank you!